internal/backend/runtime/omni/controllers/omni/talos_extensions.go

```diff
@@ -52,7 +55,16 @@ func NewTalosExtensionsController(imageFactoryClient *imagefactory.Client) *Talo
 
 				// skip fetching Talos extensions for a version which isn't registered in the image factory
 				if !hasVersion {
-					return xerrors.NewTaggedf[qtransform.SkipReconcileTag]("version %q is not registered in the image factory", version.Metadata().ID())
+					v, err := semver.ParseTolerant(version.Metadata().ID())
```

Comment:
I would rather fix this in the Image Factory if that is actually the case. I don't think Image Factory reports empty result, but should report an error actually

Reply:
But how would that help? There are no errors, but it's rather the version is not yet in the image factory.

Reply:
But the `Versions` call won't fail, no?
The problem is that the versions list doesn't have the new version of Talos, so the controller skips it.

Reply:
What should happen is that `Versions` would report v1.8.0 at some point, but fetching extensions would fail, and after some time once the release is finalized, the extensions will appear, but they should never be empty.

Reply:
It will, but the old version of controller will skip reconciliation and won't trigger an update until the instance gets restarted.
It won't react to the version list updates.

Reply:
I'm confused then... it should just fail and retry to fetch extensions later?

Probably I misunderstand the issue

Reply:
we shouldn't have a TalosVersion which is not in the Image Factory in the first place I guess..

Reply:
I guess this happens: 1.8.0 gets added to the list of versions in Omni, but it's not in the list of versions returned by the image Factory. Omni skips reconciliation and never comes back to check it again

Reply:
should we base Omni version on Image Factory versions? as if Image Factory doesn't report a version, anyways all operations would fail

Reply:
Well, yeah, that's possible.